src/rgw/rgw_role.h

```diff
@@ -0,0 +1,111 @@
+#ifndef CEPH_RGW_ROLE_H
+#define CEPH_RGW_ROLE_H
+
+class RGWRole
+{
+  static const string role_name_oid_prefix;
+  static const string role_oid_prefix;
+  static const string role_path_oid_prefix;
+  static const string role_arn_prefix;
+
+  CephContext *cct;
+  RGWRados *store;
+  string id;
+  string name;
+  string path;
+  string arn;
+  string creation_date;
+  string trust_policy;
+  map<string, string> perm_policy_map;
```

Comment:
@adamemerson : I have already added a string to hold the assume-role-policy (trust_policy here) and a map of permission policies (access policies). A role can have more than one access policies attached and also the operations for access policies are such that we need an association between the policy name and the actual policy. 

Is there anything else that you want me to add.